### ISSUE TITLE
how to access form without authentication

### ISSUE THREAD

**Issue by man-chi** [AOT-Technologies/forms-flow-ai#150]
right now, if accessing port 3000, authentication is always required. but in the case of public form access,  it is common to have the form available without login first (a.k.a guest user or anonymous user) . may I know how to setup form access without authentication required, or else, can we create a guest user in keycloak?

**Comment by abhilash-aot** [comment]
@man-chi 

You need to update form submission react route (form/:formid/) at forms-flow-web or add a public route pointing to the same component, update component a bit to work without authorisation or temp authorisation (with generation of a forms-flow-forms token related to anonymous forms-flow-forms id) and also add forms-flow-forms anonymous role for giving create submission permission while saving the forms at designer level.

Refer these for implementation:
1. https://github.com/AOT-Technologies/forms-flow-ai/blob/master/forms-flow-web/src/components/PublicRoute.jsx
2. https://github.com/AOT-Technologies/forms-flow-ai/blob/1735fdef8d49227369fa753da12c2d0cff8f5188/forms-flow-web/src/services/UserService.js#L111
3. https://github.com/AOT-Technologies/forms-flow-ai/blob/1735fdef8d49227369fa753da12c2d0cff8f5188/forms-flow-web/src/constants/constants.js#L161

We have not released this specific scenario onto Opensource yet but can be easily customisable. We did a sample before for another client for submitting forms as anonymous user .

**Comment by DivyaKMenon** [comment]
@abhilash-aot Thanks to you. Your suggestion worked. We were able to publish the form publically. But we are not able to view the anonymous submissions in forms-flow-web UI. Is there anything else we will have to add like permissions/roles to enable it?

**Comment by DivyaKMenon** [comment]
@abhilash-aot From PublicRoute.js only formiotoken is set and not the Keycloak token (authToken). Should we be setting that for an anonymous user? The API in 5000 port is expecting an authToken to be sent for CREATE but the same is not provided from the web while calling the API.

**Comment by DivyaKMenon** [comment]
@abhilash-aot The Bearer token passed from the form is showing undefined. Please let us know if we have missed out something. Adding a screenshot for the API.
<img width="1603" alt="Screenshot 2021-10-05 at 12 34 06 PM" src="https://user-images.githubusercontent.com/9332986/135976268-e77bb09c-9edf-467c-84de-b0cdec4789bc.png">

**Comment by DivyaKMenon** [comment]
@abhilash-aot The Bearer token passed from the form is showing undefined. Please let us know if we have missed out something. Adding a screenshot for the API.
<img width="1603" alt="Screenshot 2021-10-05 at 12 34 06 PM" src="https://user-images.githubusercontent.com/9332986/135976880-5a416fa2-ed4e-454f-ac04-d376911af3e9.png">

**Comment by abhilash-aot** [comment]
@DivyaKMenon  You are right we need to have an Anonymous User in Keycloak/ temporary created keycloak token too for achieving this. Temporary keycloack token can be generated from grant type like using client id and pass it. Would need to  tweak the userservice.js code too adding an anonymousKeycloakAuthInit function which calls the keycloak login api and then authenticates instead of initKeycloak https://github.com/AOT-Technologies/forms-flow-ai/blob/1735fdef8d49227369fa753da12c2d0cff8f5188/forms-flow-web/src/services/UserService.js#L27

**Comment by DivyaKMenon** [comment]
Okay. Thank you got your point. Will check and update you in case of any queries.
